tests/UKHO.ERPFacade.API.UnitTests/Controllers/WebhookControllerTests.cs

```diff
@@ -0,0 +1,76 @@
+using UKHO.ERPFacade.Common.Logging;
+
+namespace UKHO.ERPFacade.API.UnitTests.Controllers
+{
+    [TestFixture]
+    public class WebhookControllerTests
+    {
+        private IHttpContextAccessor _fakeHttpContextAccessor;
+        private ILogger<WebhookController> _fakeLogger;
+
+        private WebhookController _fakeWebHookController;
+
+        [SetUp]
+        public void Setup()
+        {
+            _fakeHttpContextAccessor = A.Fake<IHttpContextAccessor>();
+            _fakeLogger = A.Fake<ILogger<WebhookController>>();
+
+            _fakeWebHookController = new WebhookController(_fakeHttpContextAccessor, _fakeLogger);
+        }
+
+        [Test]
+        public void TestDoesWebhookReturns200OkResponseWhenValidHeaderRequestedInNewEncContentPublishedEventOptions()
+        {
+            var responseHeaders = A.Fake<IHeaderDictionary>();
+            var httpContext = A.Fake<HttpContext>();
+
+            A.CallTo(() => httpContext.Response.Headers).Returns(responseHeaders);
+            A.CallTo(() => _fakeHttpContextAccessor.HttpContext).Returns(httpContext);
+            A.CallTo(() => httpContext.Request.Headers["WebHook-Request-Origin"]).Returns(new[] { "test.com" });
+
+            var result = (OkObjectResult)_fakeWebHookController.NewEncContentPublishedEventOptions();
+
+            result.StatusCode.Should().Be(200);
+
+            A.CallTo(_fakeLogger).Where(call => call.Method.Name == "Log"
+              && call.GetArgument<LogLevel>(0) == LogLevel.Information
+              && call.GetArgument<EventId>(1) == EventIds.NewEncContentPublishedEventOptionsCallStarted.ToEventId()
+              && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Started processing the Options request for the New ENC Content Published event for webhook. | WebHook-Request-Origin : {webhookRequestOrigin} | _X-Correlation-ID : {CorrelationId}").MustHaveHappenedOnceExactly();
+
+            A.CallTo(_fakeLogger).Where(call => call.Method.Name == "Log"
+              && call.GetArgument<LogLevel>(0) == LogLevel.Information
+              && call.GetArgument<EventId>(1) == EventIds.NewEncContentPublishedEventOptionsCallCompleted.ToEventId()
+              && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Completed processing the Options request for the New ENC Content Published event for webhook. | WebHook-Request-Origin : {webhookRequestOrigin} | _X-Correlation-ID : {CorrelationId}").MustHaveHappenedOnceExactly();
+
+            A.CallTo(() => responseHeaders.Add("WebHook-Allowed-Rate", "*")).MustHaveHappened();
+            A.CallTo(() => responseHeaders.Add("WebHook-Allowed-Origin", "test.com")).MustHaveHappened();
+        }
+
+        [Test]
+        public async Task TestDoesWebhookReturns200OkResponseWhenValidNewEncContentPublishedEventReceived()
```

Comment:
Similarly, can the test case TestDoesWebhookReturns200OkResponseWhenValidNewEncContentPublishedEventReceived be renamed as WhenValidEventInNewEncContentPublishedEventReceived_ThenWebhookReturns200OkResponse

Reply:
renaming done.